netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsHandler.java

```diff
@@ -291,7 +291,15 @@ public void run() {
     }
 
     private void completeBody(final ChannelHandlerContext ctx) {
-        ctx.pipeline().remove(ctx.name() + "-body-subscriber");
+        // Netty 4.x will schedule a channelInactive event, which eventually
+        // will call a `DefaultChannelPipeline.destroy` which will remove all
+        // the handlers from the pipeline, so we need to check if the handler
+        // is still inside our pipeline
+        // See https://github.com/playframework/playframework/issues/6151
+        String channelName = ctx.name() + "-body-subscriber";
+        if (ctx.pipeline().get(channelName) != null) {
+            ctx.pipeline().remove(channelName);
```

Comment:
Does this actually fix the race condition?  I think it's possible that the channel could be removed between when we get it, and when we try to remove it.